### DISCUSSION TITLE
Dynamic height

### DISCUSSION THREAD

**Discussion by lilouartz** [vercel/satori#614] [General]
Satori asks to provide width and height of the image. What if the image height is dynamic?

The image is a of table of ingredients.

I want to render this as a table https://pillser.com/supplements/michael-s-naturopathic-kidney-factors-403

**Comment by jpope19** [comment]
has anyone figured this out yet?

**Comment by edenstrom** [comment]
Hey! I've solved this in our project. What it basically boils down to is to render twice.

1. Render it normally
2. Read the height from the first mask>rect.
   See image below that the height is 752 in a 400 container.
4. Render it again but with the new height.

> [!IMPORTANT]  
> If it doesn't work for you. Make sure that it overflows with flex and do not use `height: 100%` for the root element.

[Playground](https://satori-playground.vercel.app/?share=jVXLjttGEPyVXp4JxTF8CATHi4XtAAFsI0ic216anJbUAYfDncc-sth_d_WMKMk-WReRmH5UV1UPn7sxOOm23Vun97cz1V_KT5P8_vxMTtMy8dOWbrvdJI-3XU_2_0GjjFnDbAdjmIqf7WjPy5Z-fd3Tws7pvLcXenlpVd-txa3RuQOyQ7QyDwfNctsh_t1nyew489tfEPqTeQAW5vyP_i9bev3GqnwKUTzpkoonZ9GUNBN7yUieEwaQXCKx00WTjsBLMmne0J-Z74pQ4kVlzkLCESXuSkjE41gS-5LoHlMvme34XmZF0doDHVvBUSkhJDNpSTn0FGUpTnl2LKjFSXvyvJ_ZU5a5IhkmnGpWTZTEXdEN-TBJysrSk04TilT0x0aGEUVlmjhfVu-tvKdZRkmYmLMOhldiICdJYpnzcaQkd0WJhw29D94Hp-TKqDadPEoca26YMVLFS8ky1eEf0y0rl1YY8JbEdR5FsKYNfbQOfWP6riATA-7KXhkE3yRoIsC6UrRSeKLuUiAnEx4ByvOjermmmzia5mO-GOgsK7A2WavcOo_qENAjdAC3jYhr-oKg1B8ZiAL5gceIX0VvzWwYsOfA52lqp5gSzHr4QWdQtUw68hAsFmpDJyjqlxAr71pireLMi47CbodeaCQxwpHW3etc0jV9QgmEhgWcE-jwp1jInk3sVpUJ3ZUGwYJAa1gEpkVjIxc8DMRpkaiolY5OWb2arujrGRi8vreZTVyEQGwMbVC-78MlE5wBAg86NlwrE1f0R-2JuwBLUY192opEk82jAdtCPKlZckP_zk4I6HYSG91LMCINzeqAM3gcqCXKhv4-CzQX-J0ES_VdoUkHM_hF_9PuNBZg97YsiDe6P59XawwxIssW0nagKjgeF6J6HolRDkgEtEw7HtG1CdsIviCTzxB827kLSPUKqSnX9OEI6mwwY9lCYBIP6k-GbcIa-xD3h61eabukAoYUlGJgOdELFYEnMRiAm9Yta6Rd03vgsHHqLYPns6Wt68VAK6dHULgqbe3Ry-xpPoq4iqrwoOZ4laxuqnM30DUc5NQxUm0Cl0nAtfGlbho4P6nfeGG3ob8mMM9wYrvT1zmvLj8T7bnru7pHc-q2z92Dunzotr-9etV3B9H9IXfbN_YChsu-2-ZYpO_Eh__069Ni38L8UN9Qxj4qH_0groW99F3mAREHOCk8hDi57uUb)

![CleanShot 2024-11-27 at 19 35 01@2x](https://github.com/user-attachments/assets/3df099f4-7e00-47aa-ac01-efa838e3e8f9)

After updating the height to 752, it fits perfectly.

![CleanShot 2024-11-27 at 19 38 52@2x](https://github.com/user-attachments/assets/0c3bab53-f46f-4180-a367-f50298680049)

**Comment by aryanbaburajan** [comment]
Several hours of agony later, I realized that you can just pass the height as `undefined` and it automatically calculates it for you. I wish this was mentioned in the documentation ~~it would've saved me hours~~ but maybe I missed it.
